### PR TITLE
fix: implement missing-strategy zero for imputation

### LINKED ISSUE #597: fix: --missing-strategy zero advertised but not implemented
## Bug Description

The CLI advertises `--missing-strategy zero` as a valid option but fails at runtime with "unsupported missing value strategy: zero".

## Evidence

- `internal/cobra/analyze.go:278`: Error message suggests "drop, mean, median, zero" as valid options
- `internal/cobra/analyze.go:288`: Validation accepts "zero" as valid
- `internal/core/missing_handler.go:35-47`: Implementation only has switch cases for drop, mean, median - missing zero case

## Impact

Users following the error message guidance will hit an error instead of getting their data imputed with zeros.

## Expected Behavior

When `--missing-strategy zero` is specified, missing values should be replaced with 0.0.

## Fix

Add a case to the switch statement in `HandleMissingValues`:

```go
case types.MissingZero:
    return h.imputeWithZero(data, missingInfo, selectedCols)
```

And implement the `imputeWithZero` method similar to `imputeWithMean`.

### PR DESCRIPTION
**Pull request by bitjungle**
## Summary
Implements the advertised but missing `--missing-strategy zero` option that was causing runtime errors when users tried to use it.

## Changes
- ✅ Add `MissingZero` constant to `pkg/types/pca.go`
- ✅ Implement `imputeWithZero()` method in `internal/core/missing_handler.go`
- ✅ Add switch case for `types.MissingZero` in `HandleMissingValues`
- ✅ Update JSON schema to include "zero" in `MissingValueStrategy` enum
- ✅ Add "Impute with Zero" option to Desktop UI dropdown
- ✅ Add comprehensive test for zero imputation strategy

## Testing
- All unit tests pass including new zero imputation test
- No regressions detected across all missing value strategies
- Core package maintains 86.1% coverage
- Pre-commit hooks passed (fmt, vet, tests)
- Linters passed with no new issues

## Impact
- **No breaking changes** - purely additive feature
- Brings CLI and Desktop UI into sync
- Resolves runtime errors when users specify `--missing-strategy zero`

Closes #597

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
...
+## [1.1.7] - 2025-11-07
+
+### Fixed
+- RFC 4180 compliant CSV field escaping to prevent data corruption (#593, #588)
+- DocumentationViewer modal state reset to prevent stale content flash (#595, #589)
+- GoCSV async history clearing with proper await (#592, #587)
+- CLI command generator uses actual file paths instead of dataset names (#590, #585)
+
+### Changed
+- Removed dead code for non-existent kernel types (code cleanup) (#591, #586)
```
[the remark this answers is not in the document]
Comment (bitjungle) [reply CHANGELOG.md]: This addition will be in 1.1.8
